### PR TITLE
Added ILabelSymbol properties to loops and try statements for exposur…

### PR DESCRIPTION
**Pull request by 333fred**
Adds ILabelSymbol properties to loops and try operations for implicit continue and exit, as appropriate. For test verification, I took the approach of printing standard strings for continue/exit where they exist only, to minimize diff in unrelated C# changes. Tagging @AlekseyTs @dotnet/roslyn-compiler @dotnet/analyzer-ioperation for review. Fixes https://github.com/dotnet/roslyn/issues/23780.

### CODE REVIEW

File: src/Compilers/Core/Portable/Operations/ITryOperation.cs
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/Operations/ITryOperation.cs]: >ILabelSymbol ExitLabel { get; } [](start = 8, length = 31)

It looks like also we need to add exit label for switch (Select Case in VB).

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs
Comment (AlekseyTs) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: >propertyStringBuilder.Append(", ContinueLabel"); [](start = 16, length = 48)

I think we should at least include label's name as in VisitBranch(IBranchOperation operation).

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs
Comment (AlekseyTs) [inline src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: >LogString($" (ExitLabel)"); [](start = 16, length = 27)

Same comment here

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs
Comment (AlekseyTs) [reply src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: Also, given that the names of the labels often not unique, I think we should assign some kind of unique id to each label and include it along with the name or instead of the name. This applies to all nodes that expose label symbols. Probably can be done separately to avoid massive baseline changes for tests unrelated to the scenarios primarily targeted by this PR. 

---
In reply to: [169866369](https://github.com/dotnet/roslyn/pull/24979#discussion_r169866369) [](ancestors = 169866369)

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs
Comment (mavasani) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Why are these null instead of continue and break labels on [BoundLoopStatement](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Generated/BoundNodes.xml.Generated.cs,ed594aa427a9ad24)?

Comment (333fred) [reply src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Because I had a foolish notion that you have disabused me of.

---
In reply to: [170057138](https://github.com/dotnet/roslyn/pull/24979#discussion_r170057138) [](ancestors = 170057138)

File: src/Compilers/Core/Portable/Operations/ITryOperation.cs
Comment (333fred) [reply src/Compilers/Core/Portable/Operations/ITryOperation.cs]: Added.

---
In reply to: [169865494](https://github.com/dotnet/roslyn/pull/24979#discussion_r169865494) [](ancestors = 169865494)

File: src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs
Comment (333fred) [reply src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs]: I switched to generating ids, as C#'s generated label names change every run.

---
In reply to: [169867028](https://github.com/dotnet/roslyn/pull/24979#discussion_r169867028) [](ancestors = 169867028,169866369)

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs
Comment (333fred) [reply src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: Added C# support.

---
In reply to: [170062363](https://github.com/dotnet/roslyn/pull/24979#discussion_r170062363) [](ancestors = 170062363,170057138)

**Comment by 333fred** [comment]
@AlekseyTs @mavasani addressed feedback.

**Review by mavasani — APPROVED** [review]

**Comment by AlekseyTs** [comment]
Please add reference to the issue this PR is addressing. Also, please add a test for the scenario from the issue.

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 2), modulo the test suggestion.

**Comment by 333fred** [comment]
Added.

---
In reply to: [368094283](https://github.com/dotnet/roslyn/pull/24979#issuecomment-368094283) [](ancestors = 368094283)

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler @cston I need a second review, please.

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 3)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off

**Comment by AlekseyTs** [comment]
@333fred It looks like more tests need an adjustment

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 4)

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler I need a second review please.

**Comment by 333fred** [comment]
@dotnet-bot retest windows_release_vs-integration_prtest please

**Review by gafter — APPROVED** [review]
:shipit:

**Pull request merged**
